src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,100 @@
+
 public class WorkWithFile {
+
+    private static final String SPLIT_REGEX = "\\r?\\n";
+    private static final String SEP = ",";
+    private static final String BUY = "buy";
+    private static final String SUPPLY = "supply";
+    private static final String RESULT = "result";
+    private static final String NEW_LINE = System.lineSeparator();
+
+    private static final class Totals {
+        private final int supply;
+        private final int buy;
+
+        Totals(int supply, int buy) {
+            this.supply = supply;
+            this.buy = buy;
+        }
+    }
+
     public void getStatistic(String fromFileName, String toFileName) {
 
+        String content = readFile(fromFileName);
+        Totals totals = parseAndAggregate(content);
+        String report = buildReport(totals);
+        writeFile(toFileName, report);
+    }
+
+    private String readFile(String fromFileName) {
+        try {
+            return Files.readString(Path.of(fromFileName));
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data from a file: " + fromFileName + e);
+        }
+    }
+
+    private Totals parseAndAggregate(String content) {
+        int sumSupply = 0;
+        int sumBuy = 0;
+
+        String[] lines = content.split(SPLIT_REGEX);
+        for (int i = 0; i < lines.length; i++) {
+            String line = lines[i].trim();
+            if (line.isEmpty()) {
+                continue;
+            }
+
+            String[] parts = line.split(SEP);
+            if (parts.length != 2) {
+                continue;
+            }
+
+            String op = parts[0].trim().toLowerCase();
+            String amountStr = parts[1].trim();
+
+            if (!SUPPLY.equals(op) && !BUY.equals(op)) {
+                continue;
+            }
+
+            int amount;
+            try {
+                amount = Integer.parseInt(amountStr);
+            } catch (NumberFormatException ex) {
+                continue;
```

This violates checklist items "Amount must be parseable as an integer." and "Don't ignore exceptions". The NumberFormatException is caught and ignored (`continue`), which hides malformed numeric input and can produce incorrect totals. Replace the silent swallow with a rethrow that includes context, e.g.:
```
catch (NumberFormatException ex) {
    throw new RuntimeException("Can't parse amount '" + amountStr + "' in line: '" + line + "'", ex);
}
```
So tests and callers get a clear deterministic failure.